src/eventSchemas/templates/schemasAppTemplateUtils.ts

```diff
@@ -0,0 +1,114 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import _ = require('lodash')
+import { IdentifierFormatter, SchemaCodeGenUtils } from '../../eventSchemas/models/schemaCodeGenUtils'
+import { SchemaClient } from '../../shared/clients/schemaClient'
+
+const X_AMAZON_EVENT_SOURCE = 'x-amazon-events-source'
+const X_AMAZON_EVENT_DETAIL_TYPE = 'x-amazon-events-detail-type'
+
+const COMPONENTS = 'components'
+const SCHEMAS = 'schemas'
+const COMPONENTS_SCHEMAS_PATH = '#/components/schemas/'
+const AWS_EVENT = 'AWSEvent'
+const PROPERTIES = 'properties'
+const DETAIL = 'detail'
+const REF = '$ref'
+
+const DEFAULT_EVENT_SOURCE = 'INSERT-YOUR-EVENT-SOURCE'
+const DEFAULT_EVENT_DETAIL_TYPE = 'INSERT-YOUR-DETAIL-TYPE'
+
+export interface SchemaTemplateParameters {
+    SchemaVersion: string
+    templateExtraContent: SchemaTemplateExtraContext
+}
+
+// This matches the extra_content parameters to Schema-based templates in both key and value names in cookiecutter.json in the templates used by SamEventBridgeSchemaAppPython
+export interface SchemaTemplateExtraContext {
+    AWS_Schema_registry: string
+    AWS_Schema_name: string
+    AWS_Schema_root: string
+    AWS_Schema_source: string
+    AWS_Schema_detail_type: string
+    user_agent: string
+}
+
+export async function buildSchemaTemplateParameters(schemaName: string, registryName: string, client: SchemaClient) {
+    const response = await client.describeSchema(registryName, schemaName)
+    const schemaNode = JSON.parse(response.Content!)
+    const latestSchemaVersion = response.SchemaVersion
+    // Standard OpenAPI specification for AwsEventNode
+    const awsEventNode = _.get(schemaNode, COMPONENTS.concat('.', SCHEMAS, '.', AWS_EVENT))
+
+    // Derive source from custom OpenAPI metadata provided by Schemas service
+    let source = _.get(awsEventNode, X_AMAZON_EVENT_SOURCE)
+    if (!_.isString(source)) {
+        source = DEFAULT_EVENT_SOURCE
+    }
+
+    // Derive detail type from custom OpenAPI metadata provided by Schemas service
+    let detailType = _.get(awsEventNode, X_AMAZON_EVENT_DETAIL_TYPE)
+    if (!_.isString(detailType)) {
+        detailType = DEFAULT_EVENT_DETAIL_TYPE
+    }
```

Comment:
I feel like there's a better way to get this to enforce that this is a string (to eliminate the `no-unsafe-any` underneath). What non-string values can be gotten from `_.get`?

Reply:
In reality  `_.get(awsEventNode, X_AMAZON_EVENT_DETAIL_TYPE)` would either return undefined or it would return string. If we look at all 1p schemas( aws.events), they all have string value in that nested json key chain.  However, compiler can't know what type it would be, thus returns any

Originally I had it 
 `_.get(awsEventNode, X_AMAZON_EVENT_DETAIL_TYPE,DEFAULT_EVENT_DETAIL_TYPE)` , however when I was passing this value I had to add  `as string` to silence warnings.

Reply:
Try:

` const detailType: string = _.get(awsEventNode, X_AMAZON_EVENT_DETAIL_TYPE, DEFAULT_EVENT_DETAIL_TYPE)`

Reply:
This one still gives the same  warning `Unsafe use of expression of type 'any'. (no-unsafe-any)`